### PR TITLE
Update Pywikibot to 6.6.2

### PR DESCRIPTION
**Pull request by legoktm**
Bug: T294681

### CODE REVIEW

**Review by chicocvenancio — DISMISSED** [review]

**Comment by dipietrom** [comment]
Where is the container for this? I would like to run it.

**Review by dipietrom — APPROVED** [review]

**Review by dipietrom — CHANGES_REQUESTED** [review]
To get us away from using latest the current setup is using YY.DD.patch versioning. So we'll probably need to update values.yaml for this to go out. Either back to latest. Or we can update to a new tag in values and tag the container upstream. I'll look into getting travis to do more of this.

**Comment by legoktm** [comment]
@dipietrom please amend/fix as you see fit, this is the first time we're explicitly upgrading Pywikibot like this since 5c51ec49f59fd33d1a415f2ee930d98c1c09c900 pinned to a specific version.

**Comment by dipietrom** [comment]
I've updated the tag in values. And tested locally. I'm pushing the image, but my upload is being very slow today, so it will probably not be in quay.io for awhile. Ideally this could be done by travis, maybe we could convince travis to make a tag out of the PR number such that it is unique to each feature? I'll look more into the travis code to see what can be done. Another thing that might be reasonable, and more similar to what we were doing, is to modify the tag in values.yaml to a new tag, then update in quay.io to add a tag to the commit hash tag that travis builds, as the manually entered one. Similar to how we would move "latest" to the tag that travis builds.

Indeed I think that gets us clarity on what we've deployed and rollbacks. We are just missing the ability to spin up minikube without building a local container. Though that could be solved later, with getting travis to build on pr with something version shaped. Am I correct in that travis only pushes a build on a pr merge?

If that is the case we can just link singleuser:628aa0 (which would be built on merge) to singleuser:21.11.01 and we're done regardless of if my upload completes.

**Review by dipietrom — APPROVED** [review]

**Comment by chicocvenancio** [comment]
@dipietrom you might want to take a look at https://github.com/toolforge/paws/blob/master/build.py, it used to be called both by travis to build images as by a cluster-run container that would run it to call helm with the image values of the hashes. This was changed when we migrated PAWS from toolforge a few years back and didn't migrate the deploy-hook, moving to `latest` tags that were on the repo. 

CI it seems moving to GHA is the more straightforward path right now, #70 being a good start. 

For CD I'm not sure we want to setup the deploy-hook again. While I don't know anything that is wrong with it I don't feel too comfortable rolling our own implementation on this. We're deploying rarely enough that it is not a huge issue, but maybe having a small argocd install for this would make sense. 


TL;DR `build.py` does both CI and CD; did CI when called by travis, that we probably want to move to GHA; did CD with locally calling helm[ and `--set` flags.]( https://github.com/toolforge/paws/blob/master/build.py#L102)

**Comment by dipietrom** [comment]
@chicocvenancio Thank you that is good info on what build.py is up to. I saw that travis was using it, as was tox. I pulled out travis in [#98](https://github.com/toolforge/paws/pull/98) with [#70](https://github.com/toolforge/paws/pull/70) as a helpful starting point. Though I left build.py alone, even though there are surely parts of it that are no longer used. Though tox is presumably still using it. I left a note in PR98 that we could have autodeploy, as you describe with helm and a webhook, though it is not part of that PR.
